### PR TITLE
Move infrastructure tests to our new test suite

### PR DESCRIPTION
**Pull request by remi-stripe**
Continue moving more tests to the new test suite

Focused on Infrastructure tests for now
- Ensure user agent and custom headers are set properly
- Ensure simple expansion is supported on a resource and on a list
- Ensure that multi-level expansion is supported
- Ensure parameter encoding works as expected

cc @ob-stripe to have a look
cc @stripe/api-libraries

### CODE REVIEW

File: src/StripeTests/Infrastructure/test_data/TestService.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+namespace StripeTests.Infrastructure.test_data
```
Comment (ob-stripe) [inline src/StripeTests/Infrastructure/test_data/TestService.cs]: Can you rename this to `TestData` instead (and also change the directory's name)? I don't think the use of snake case in namespaces is idiomatic in C#.

Comment (remi-stripe) [reply src/StripeTests/Infrastructure/test_data/TestService.cs]: fixed!

**Comment by remi-stripe** [comment]
@ob-stripe Could I ask you to have a look? We're almost there, I need to add tests for Cards, Events and one or two encoding logic and we're done.

**Comment by remi-stripe** [comment]
Okay this work is complete. The old test suites have been removed, all existing tests have been ported to the new test suite and it can now be reviewed and merged.

r? @ob-stripe

**Comment by ob-stripe** [comment]
Pulled this in. Once again, amazing work @remi-stripe 🎉

**Pull request merged**